src/Database/QueryCompiler.php

```diff
@@ -195,14 +185,17 @@ protected function _buildWithPart(array $parts, Query $query, ValueBinder $binde
      */
     protected function _buildSelectPart(array $parts, Query $query, ValueBinder $binder): string
     {
+        $driver = $query->getConnection()->getDriver($query->getConnectionRole());
         $select = 'SELECT%s %s%s';
-        if ($this->_orderedUnion && $query->clause('union')) {
+        if (
+            $driver->supports(DriverFeatureEnum::SET_OPERATIONS_ORDER_BY) &&
```

Comment:
Nitpick: I would check the clause first before checking if the feature is supported.

Reply:
Makes sense. Changed it!